cmd/thanos/tools_bucket.go

```diff
@@ -1165,7 +1164,15 @@ func registerBucketRewrite(app extkingpin.AppClause, objStoreConfig *extflag.Pat
 				if err != nil {
 					return errors.Wrapf(err, "read meta of %v", id)
 				}
-				b, err := tsdb.OpenBlock(logger, filepath.Join(tbc.tmpDir, id.String()), chunkPool)
+
+				var pool chunkenc.Pool
+				if meta.Thanos.Downsample.Resolution == 0 {
+					pool = chunkenc.NewPool()
```

Comment:
Shouldn't this pool be shared among different `tsdb.OpenBlock` to reduce allocations? I think you should create two pools and then pass either `downsample.Pool` or `chunkenc.Pool` depending on the resolution. Even though we have no parallelism here at the moment but I think it still makes sense to use `sync.Pool` properly because the allocator might not free items from `sync.Pool` between iterations.

Reply:
We only create one pool per block, which should be on the order of tens, maybe low hundreds. Not sure how much practical savings we can get from optimizing there.

That said, I'm of course happy to make the change if requested. 👍

Reply:
IMHO we will probably add concurrency here in the future so maybe it makes sense to do this since the beginning? :smile: